kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/main/java/org/kie/workbench/common/stunner/bpmn/definition/property/diagram/AbstractDataTypeCache.java

```diff
@@ -0,0 +1,191 @@
+            IntermediateEscalationEvent bt = (IntermediateEscalationEvent) view.getDefinition();
+            allDataTypes.addAll(processAssignments(bt.getDataIOSet().getAssignmentsinfo()));
+        } else if (definition instanceof IntermediateEscalationEventThrowing) {
+            IntermediateEscalationEventThrowing bt = (IntermediateEscalationEventThrowing) view.getDefinition();
+            allDataTypes.addAll(processAssignments(bt.getDataIOSet().getAssignmentsinfo()));
+        } else if (definition instanceof IntermediateMessageEventCatching) {
+            IntermediateMessageEventCatching bt = (IntermediateMessageEventCatching) view.getDefinition();
+            allDataTypes.addAll(processAssignments(bt.getDataIOSet().getAssignmentsinfo()));
+        } else if (definition instanceof IntermediateMessageEventThrowing) {
+            IntermediateMessageEventThrowing bt = (IntermediateMessageEventThrowing) view.getDefinition();
+            allDataTypes.addAll(processAssignments(bt.getDataIOSet().getAssignmentsinfo()));
+        } else if (definition instanceof IntermediateSignalEventCatching) {
+            IntermediateSignalEventCatching bt = (IntermediateSignalEventCatching) view.getDefinition();
+            allDataTypes.addAll(processAssignments(bt.getDataIOSet().getAssignmentsinfo()));
+        } else if (definition instanceof IntermediateSignalEventThrowing) {
+            IntermediateSignalEventThrowing bt = (IntermediateSignalEventThrowing) view.getDefinition();
+            allDataTypes.addAll(processAssignments(bt.getDataIOSet().getAssignmentsinfo()));
+        } else if (definition instanceof ReusableSubprocess) {
+            ReusableSubprocess bt = (ReusableSubprocess) view.getDefinition();
+            allDataTypes.addAll(processAssignments(bt.getDataIOSet().getAssignmentsinfo()));
+        } else if (definition instanceof StartErrorEvent) {
+            StartErrorEvent bt = (StartErrorEvent) view.getDefinition();
+            allDataTypes.addAll(processAssignments(bt.getDataIOSet().getAssignmentsinfo()));
+        } else if (definition instanceof StartEscalationEvent) {
+            StartEscalationEvent bt = (StartEscalationEvent) view.getDefinition();
+            allDataTypes.addAll(processAssignments(bt.getDataIOSet().getAssignmentsinfo()));
+        } else if (definition instanceof StartMessageEvent) {
+            StartMessageEvent bt = (StartMessageEvent) view.getDefinition();
+            allDataTypes.addAll(processAssignments(bt.getDataIOSet().getAssignmentsinfo()));
+        } else if (definition instanceof StartSignalEvent) {
+            StartSignalEvent bt = (StartSignalEvent) view.getDefinition();
+            allDataTypes.addAll(processAssignments(bt.getDataIOSet().getAssignmentsinfo()));
+        } else if (definition instanceof CustomTask) {
+            CustomTask bt = (CustomTask) view.getDefinition();
+            allDataTypes.addAll(processAssignments(bt.getDataIOSet().getAssignmentsinfo()));
+        }
```

Comment:
One suggestion here, let me know if it valid. I understand that if we add some new instance and we forget to add it here we won't be able to extract the data types from it.

```suggestion
        } else {
            // log error that we tried to extract data type for a definition we did not know about?
        }
```

Reply:
Hi @domhanak basically what it does it goes through all the nodes and checks instances, if it is not from the list it does not do anything, so if it reaches else, it is not really an error, it is just a node that does not contain data types

Reply:
Maybe some info in that case?